app/assets/javascripts/discourse/app/widgets/topic-timeline.js

```diff
@@ -598,4 +599,26 @@ export default createWidget("topic-timeline", {
 
     return result;
   },
+
+  didRenderWidget() {
+    showPopup({
+      currentUser: this.currentUser,
```

Comment:
Just a thought, but it may be cleaner to check whether we should show a popup here before showing it. That way the intent is cleaner and we probably wouldn't need to pass the current user to `showPopup`.

Reply:
`currentUser` is used to update the user when the popup is dismissed too. What I wanted to achieve by `showPopup` is use as little code as possible to display popups. If I move the logic outside of this function, then I will make it slightly more difficult to show popups and duplicate the check for every popup.

Reply:
Another option is to look up the currentUser service in lib/popup. Avoids some extra lines when invoking it.

Reply:
To be able to do that I need to have a reference to the lookup function or to the container. The popup library is very isolated from all that, so I still have to pass something to it.

Reply:
I understand what you mean. I will split this logic in two: one part is going to be in the user model which performs the seen check and then call the logic that displays the popup from lib.

Reply:
Cool, I like this approach. (Note that now there's no need to explicitly pass `currentUser: this.currentUser`.)